packages/node_modules/pouchdb-abstract-mapreduce/src/index.js

```diff
@@ -811,14 +812,13 @@ function createAbstractMapReduce(localDocName, mapper, reducer, ddocValidator) {
   }
 
   function queryPromised(db, fun, opts) {
-    if (db.type() === 'http') {
-      return httpQuery(db, fun, opts);
-    }
-
     /* istanbul ignore next */
     if (typeof db._query === 'function') {
       return customQuery(db, fun, opts);
     }
+    if (isRemote(db)) {
+      return httpQuery(db, fun, opts);
+    }
```

Comment:
this ought to come after custom queries/viewCleanups because adapters can override this regardless of whether or not they're remote.